image.go

```diff
@@ -459,11 +459,12 @@ func (i *image) getManifestPlatform(ctx context.Context, manifest ocispec.Manife
 		return ocispec.Platform{}, err
 	}
 
-	var image ocispec.Image
-	if err := json.Unmarshal(p, &image); err != nil {
+	// technically, this should be ocispec.Image, but ocispec.Platform has all the same fields with the same names (and the same "json" tags), and is the data we actually want (and "json.Unmarshal" will ignore everything else anyhow)
```

Comment:
I guess we can simplify these comments now that `ocispec.Platform` is actually embedded in the image. e.g. something like;

```suggestion
	// Technically, this should be ocispec.Image, but we only need the
	// ocispec.Platform that is embedded in the image struct.
```